File: QA-project/backend/app/models/ai_queries.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.database import Base


class AIQuery(Base):
    __tablename__ = "ai_queries"

    query_id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.user_id", ondelete="CASCADE"), nullable=False)
    dataset_id = Column(Integer, ForeignKey("datasets.dataset_id", ondelete="CASCADE"), nullable=False)
    question = Column(Text, nullable=False)
    answer = Column(Text)
    source_reference = Column(String(1000))
    asked_at = Column(DateTime, server_default=func.now(), nullable=False)

    # Relationships
    user = relationship("User", back_populates="ai_queries")
    dataset = relationship("Dataset", back_populates="ai_queries")

File: QA-project/backend/src/db/schema/ai-queries.ts
import {
  pgTable,
  serial,
  integer,
  text,
  varchar,
  timestamp,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { users } from "./users";
import { datasets } from "./datasets";

export const aiQueries = pgTable("ai_queries", {
  queryId: serial("query_id").primaryKey(),
  userId: integer("user_id")
    .notNull()
    .references(() => users.userId, { onDelete: "cascade" }),
  datasetId: integer("dataset_id")
    .notNull()
    .references(() => datasets.datasetId, { onDelete: "cascade" }),
  question: text("question").notNull(),
  answer: text("answer"),
  sourceReference: varchar("source_reference", { length: 1000 }),
  askedAt: timestamp("asked_at").defaultNow().notNull(),
});

export const aiQueriesRelations = relations(aiQueries, ({ one }) => ({
  user: one(users, {
    fields: [aiQueries.userId],
    references: [users.userId],
  }),
  dataset: one(datasets, {
    fields: [aiQueries.datasetId],
    references: [datasets.datasetId],
  }),
}));
